lib/Backend/arm64/LegalizeMD.cpp

```diff
@@ -835,6 +881,21 @@ bool LegalizeMD::LegalizeDirectBranch(IR::BranchInstr *branchInstr, uint32 branc
         {
             return false;
         }
+        limit = 0x0003ffff;
+    }
+
+    // Attempt to find an unconditional branch within the range and insert a branch island below it:
+    //  CB. $island
+    //  ...
+    //  B
+    //  $island:
+    //  B $target
```

Comment:
Is it possible to simply invert the condition and short-branch to the previous fallthrough? That may be easier in some situations.
e.g.
cb. $target
fallthroughcode
$target:
to 
cbn. $fallthrough
b $target
$fallthrough:
fallthroughcode

Reply:
I added the branch island approach at suggestion from @LouisLaf. Branch inversion is done if an island isn't found.